### PR TITLE
qcsxcad: init at 0.6.2

### PR DESCRIPTION
**Pull request by matthuszagh**
###### Motivation for this change
Part of [this pr](https://github.com/NixOS/nixpkgs/pull/69262). @jonringer 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/qcsxcad/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    tinyxml
+    vtkWithQt5
+    qt5.qtbase
+  ];
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "Qt-GUI for CSXCAD";
+    homepage = https://github.com/thliebig/QCSXCAD;
```
Comment (jonringer) [inline pkgs/applications/science/electronics/qcsxcad/default.nix:47]: rfc 45
```suggestion
    homepage = "https://github.com/thliebig/QCSXCAD";
```

File: pkgs/applications/science/electronics/qcsxcad/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, mkDerivation
+, fetchFromGitHub
+, cmake
+, csxcad
+, tinyxml
+, vtkWithQt5
+, qt5
```
Comment (jonringer) [inline pkgs/applications/science/electronics/qcsxcad/default.nix:8]: you should import the individual packages from the qt5 package set
```suggestion
, wrapQtAppsHook
```

File: pkgs/applications/science/electronics/qcsxcad/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  src = fetchFromGitHub {
+    owner = "thliebig";
+    repo = "QCSXCAD";
+    rev = "v${version}";
+    sha256 = "1l77xr1skf9rrd0pwdw0wcz8a3qzwnn9vz7613j2w5xwakagiglj";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    qt5.wrapQtAppsHook
```
Comment (jonringer) [inline pkgs/applications/science/electronics/qcsxcad/default.nix:27]:
```suggestion
    wrapQtAppsHook
```

File: pkgs/applications/science/electronics/qcsxcad/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+, vtkWithQt5
+, qt5
+}:
+
+let
+  version = "0.6.2";
+in
+mkDerivation {
+  pname = "qcsxcad";
+  version = version;
```
Comment (jonringer) [inline pkgs/applications/science/electronics/qcsxcad/default.nix:16]: I find this more readable
```suggestion
mkDerivation rec {
  pname = "qcsxcad";
  version = "0.6.2";
```

**Comment by jonringer** [comment]
Also, I don't see the gui being installed:
```
[nix-shell:/home/jon/.cache/nixpkgs-review/pr-95100]$ tree results/qcsxcad
results/qcsxcad
├── include
│   └── QCSXCAD
│       ├── export_pov.h
│       ├── export_x3d.h
│       ├── QCSGridEditor.h
│       ├── QCSPrimEditor.h
│       ├── QCSPropEditor.h
│       ├── QCSTreeWidget.h
│       ├── QCSXCAD_Global.h
│       ├── QCSXCAD.h
│       ├── QParameterGui.h
│       ├── QVTKStructure.h
│       ├── vtkInteractorStyleRubberBand2DPlane.h
│       └── VTKPrimitives.h
└── lib
    ├── libQCSXCAD.so -> libQCSXCAD.so.0
    ├── libQCSXCAD.so.0 -> libQCSXCAD.so.0.6.2
    └── libQCSXCAD.so.0.6.2

3 directories, 15 files
```

**Comment by matthuszagh** [comment]
@jonringer Are you commenting on the fact that there isn't an application binary among the outputs? [QCSXCAD](https://github.com/thliebig/QCSXCAD) is a library so I believe this is correct. When I run [AppCSXCAD](https://github.com/thliebig/AppCSXCAD) (coming up in the next PR) the gui seems to run fine. If I misunderstood your comment, would you mind clarifying?

File: pkgs/applications/science/electronics/qcsxcad/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    csxcad
+    tinyxml
+    vtkWithQt5
+    qtbase
+  ];
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "Qt-GUI for CSXCAD";
```
Comment (jonringer) [inline pkgs/applications/science/electronics/qcsxcad/default.nix:44]: the description says qt-gui for csxcad, but there's no gui, that's what I'm commenting on

Comment (matthuszagh) [reply pkgs/applications/science/electronics/qcsxcad/default.nix]: I copied this verbatim from the project's description, but I agree it's very misleading. I've changed this to `"Qt library for CSXCAD"`. Better?

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 95100 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>28 packages marked as broken and skipped:</summary>  - digitalbitbox  - glibcCross  - linuxPackages_4_4.evdi  - linuxPackages_hardkernel_4_14.bcc  - linuxPackages_hardkernel_4_14.bpftrace  - linuxPackages_hardkernel_4_14.can-isotp  - linuxPackages_hardkernel_4_14.digimend  - linuxPackages_hardkernel_4_14.evdi  - linuxPackages_hardkernel_4_14.mba6x_bl  - linuxPackages_hardkernel_4_14.nvidia_x11  - linuxPackages_hardkernel_4_14.r8125  - linuxPackages_hardkernel_4_14.rtl8821au  - linuxPackages_hardkernel_4_14.rtl88x2bu  - linuxPackages_hardkernel_4_14.rtl88xxau-aircrack  - linuxPackages_hardkernel_4_14.rtlwifi_new  - newlibCross  - php74Extensions.pthreads  - python27Packages.caffe  - python27Packages.habanero  - python27Packages.handout  - python27Packages.hass-nabucasa  - python27Packages.libmodulemd  - python37Packages.nixpart  - python37Packages.notify  - python38Packages.libselinux  - python38Packages.notify  - python38Packages.pyblock  - qes</details><details>  <summary>1 package built:</summary>  - qcsxcad</details>

**Pull request merged**